internal/cloud/gcp/compute.go

```diff
@@ -288,3 +315,108 @@ func (g *GCP) ComputeDiskDelete(ctx context.Context, zone, disk string) error {
 
 	return err
 }
+
+// storageRegionToComputeRegion translates a Google Storage Region to GCE Region.
+// This is useful mainly for multi and dual Storage Regions. For each valid multi
+// or dual Region name, a slice with relevant GCE Regions is returned. If the
+// Region provided as an argument is not multi or dual Region, a slice with the
+// provided argument as the only item is returned.
+//
+// In general, Storage Regions correspond to the Compute Engine Regions. However,
+// Storage allows also Multi and Dual regions, which must be mapped to GCE Regions,
+// since these can not be used with GCE API calls.
+//
+// Uses:
+//  - Compute Engine API
+func (g *GCP) storageRegionToComputeRegions(ctx context.Context, region string) ([]string, error) {
+	regionLower := strings.ToLower(region)
+
+	// Handle Dual-Regions
+	// https://cloud.google.com/storage/docs/locations#location-dr
+	if regionLower == "asia1" {
+		return []string{"asia-northeast1", "asia-northeast2"}, nil
+	} else if regionLower == "eur4" {
+		return []string{"europe-north1", "europe-west4"}, nil
+	} else if regionLower == "nam4" {
+		return []string{"us-central1", "us-east1"}, nil
+	}
+
+	// Handle Regular Region
+	if regionLower != "asia" && regionLower != "eu" && regionLower != "us" {
+		// Just return a slice with the region, which we got as
+		return []string{regionLower}, nil
+	}
+
+	// Handle Multi-Regions
+	// https://cloud.google.com/storage/docs/locations#location-mr
+	computeService, err := compute.NewService(ctx, option.WithCredentials(g.creds))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Compute Engine client: %v", err)
+	}
+
+	regionObjList, err := computeService.Regions.List(g.creds.ProjectID).Context(ctx).Do()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get available Compute Engine Regions: %v", err)
+	}
+
+	regionsMap := make(map[string][]string)
+	for _, regionObj := range regionObjList.Items {
+		regionPrefix := strings.Split(regionObj.Name, "-")[0]
+		regionsMap[regionPrefix] = append(regionsMap[regionPrefix], regionObj.Name)
+	}
+
+	switch regionLower {
+	case "asia", "us":
+		return regionsMap[regionLower], nil
+	case "eu":
+		var euRegions []string
+		for _, euRegion := range regionsMap["europe"] {
+			// "europe-west2" (London) and "europe-west6" (Zurich) are excluded
```

Comment:
why? Can you put here a link to docs?

Reply:
It is the same link as on the line 351 (https://cloud.google.com/storage/docs/locations#location-mr). I can definitely put it here as well.

As for the reasoning, these regions are not in the European Union (sorry UK :smiley:)

Reply:
I added a link below the comment.